clang/test/SemaSYCL/allow-constexpr-recursion.cpp

```diff
@@ -6,13 +6,11 @@
 
 sycl::queue q;
 
-// expected-note@+1{{function implemented using recursion declared here}}
```

Comment:
So we were only able to figure this out by pure happenstance before, there is a bug in SemaSYCL.cpp currently where we decide that the caller of a recursive function is ALSO recursive.  This is not a sound assumption, and even in the cases it is true, it doesn't catch all cases.  

Fixing that ends up meaning we cannot catch the following example, but this does not change the validity of the program (since we get 1 diagnostic here instead of 2 like previously):
```
int func1();
int func2() { func1(); }
int func1() { func2(); }
kernel<class Foo>([](){ func1(); });
```
Because we have to give up on running through diagnostics, we catch that `func1` is recursive, but cannot catch `func2`, because we don't continue running down the list.